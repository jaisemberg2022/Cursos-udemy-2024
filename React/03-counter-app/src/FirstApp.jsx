const newMessage = {
    mensaje:'Hola',
    nombre:'jaider',
};

const getSuma = (a,b) => {
    return a+b;
}

export const FirstApp = () => {

  return (
    <>
        <code>{ JSON.stringify(newMessage) }</code>
        <h4>{ getSuma(2,3) }</h4>
        <p>prueba</p>
    </>
  )
}
