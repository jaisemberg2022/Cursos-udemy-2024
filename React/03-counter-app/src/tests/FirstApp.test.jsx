import {render} from '@testing-library/react'
import {FirstApp} from '../FirstApp'
import { teal } from '@mui/material/colors';


describe('pruebas en <FirstApp>', () => { 
    // test('debe de hacer match con el snapshot', () => { 
    //     const tittle = 'Hola soy goku ';
    //     const {conatiner} = render(<FirstApp tittle={tittle}/>);
    //     expect(conatiner).toMatchSnapshot();
    //  });

     test('debe mostrar el titulo en h1', ()=>{
        const tittle = 'Hola, soy goku ';
        const {container,getByText,getByTestId} = render(<FirstApp tittle={tittle} />);

        // const h1 = container.querySelector('h1')  
        // expect(getByText(tittle)).toBeTruthy();
        // expect(h1.innerHTML).toBe(tittle);
        expect(getByTestId('testId')).toBeTruthy();
     })
 })