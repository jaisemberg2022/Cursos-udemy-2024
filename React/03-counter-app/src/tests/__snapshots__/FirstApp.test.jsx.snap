// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <FirstApp> debe de hacer match con el snapshot 1`] = `undefined`;
